controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go

```diff
@@ -228,6 +229,11 @@ func (in *KubeadmControlPlane) validateCommon() (allErrs field.ErrorList) {
 		)
 	}
 
+	_, err := semver.ParseTolerant(in.Spec.Version)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "version"), in.Spec.Version, "failed to parse kubernetes version "))
```

Comment:
What about something like this?

```suggestion
		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "version"), in.Spec.Version, "must be a valid semantic version"))
```